runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ExpressionsInRequiredExpressionsParams.java

```diff
@@ -0,0 +1,89 @@
+import static org.mule.runtime.api.meta.ExpressionSupport.REQUIRED;
+import static org.mule.runtime.ast.api.util.ComponentAstPredicatesFactory.currentElemement;
+import static org.mule.runtime.ast.api.util.ComponentAstPredicatesFactory.equalsIdentifier;
+import static org.mule.runtime.ast.api.validation.Validation.Level.ERROR;
+import static org.mule.runtime.internal.dsl.DslConstants.CORE_PREFIX;
+
+import org.mule.runtime.api.component.ComponentIdentifier;
+import org.mule.runtime.api.meta.model.parameter.ParameterizedModel;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ComponentAst;
+import org.mule.runtime.ast.api.ComponentParameterAst;
+import org.mule.runtime.ast.api.validation.Validation;
+import org.mule.runtime.extension.api.declaration.type.annotation.LiteralTypeAnnotation;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+/**
+ * No expressions are provided for parameters that do not support expressions.
+ */
+public class ExpressionsInRequiredExpressionsParams implements Validation {
+
+  private static final String FLOW_REF_ELEMENT = "flow-ref";
+
+  private static final String DEFAULT_EXPRESSION_PREFIX = "#[";
+  private static final String DEFAULT_EXPRESSION_SUFFIX = "]";
+
+  private static final ComponentIdentifier FLOW_REF_IDENTIFIER =
+      builder().namespace(CORE_PREFIX).name(FLOW_REF_ELEMENT).build();
+
+  @Override
+  public String getName() {
+    return "Expression must be set in expressionsRequired params";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Expressions are provided for parameters that require expressions.";
+  }
+
+  @Override
+  public Level getLevel() {
+    return ERROR;
+  }
+
+  @Override
+  public Predicate<List<ComponentAst>> applicable() {
+    return currentElemement(component -> component.getModel(ParameterizedModel.class).isPresent())
+        // According to the extension model, flow-ref cannot be dynamic,
+        // But this check is needed to avoid breaking on legacy cases that use dynamic flow-refs.
+        .and(currentElemement(equalsIdentifier(FLOW_REF_IDENTIFIER).negate()));
+  }
+
+  @Override
+  public Optional<String> validate(ComponentAst component, ArtifactAst artifact) {
+    for (ComponentParameterAst param : component.getParameters()) {
+      if (!param.getModel().isComponentId()
+          && param.getValue().isRight()
+          && param.getValue().getRight() instanceof String) {
+        final String stringValue = (String) param.getValue().getRight();
+
+        if (REQUIRED.equals(param.getModel().getExpressionSupport())
+            && !param.getModel().getType().getAnnotation(LiteralTypeAnnotation.class).isPresent()
+            && (!stringValue.startsWith(DEFAULT_EXPRESSION_PREFIX)
+                || !stringValue.endsWith(DEFAULT_EXPRESSION_SUFFIX))) {
+          return of(format("An expression value was not given for parameter '%s' but it requires a expression",
```

Comment:
`An expression value was not given for parameter` -> `A static value was given for parameter`